plenoptic/simulate/canonical_computations/steerable_pyramid_freq.py

```diff
@@ -321,89 +316,98 @@ def forward(self, x, scales=[]):
+                        if the complex pyramid is used, magnitudes need to be divided by sqrt(2) 
+                        because energy is doubled.
+                        '''
                         if self.tight_frame:
-                            band = band / np.sqrt(2)
-                        self.pyr_coeffs[(i, b)] = band
-                        self.pyr_size[(i, b)] = tuple(band.shape[2:4])
+                            band = band/np.sqrt(2)
+                        pyr_coeffs[(i, b)] = band
+                    self.pyr_size[(i, b)] = tuple(band.shape[-2:])
 
             if not self.downsample:
                 # no subsampling of angle and rad
                 # just use lo0mask
                 lomask = self._lomasks[i]
                 lodft = lodft * lomask
+                '''
+                because we don't subsample here, if we are not using orthonormalization that
+                we need to manually account for the subsampling, so that energy in each band remains the same
+                the energy is cut by factor of 4 so we need to scale magnitudes by factor of 2
+                '''
+                if self.fft_norm != "ortho":
+                    lodft = 2*lodft
             else:
                 # subsample indices
                 lostart, loend = self._loindices[i]
 
                 log_rad = log_rad[lostart[0]:loend[0], lostart[1]:loend[1]]
                 angle = angle[lostart[0]:loend[0], lostart[1]:loend[1]]
 
-                # subsampling
-                lodft = lodft[:, :, lostart[0]:loend[0], lostart[1]:loend[1], :]
-                # filtering
+                # subsampling of the dft for next scale
+                lodft = lodft[:, :, lostart[0]:loend[0], lostart[1]:loend[1]]
+                # low-pass filter mask is selected
                 lomask = self._lomasks[i]
-                # convolution in spatial domain
+                # again multiply dft by subsampled mask (convolution in spatial domain)
+
                 lodft = lodft * lomask
 
         if 'residual_lowpass' in scales:
             # compute residual lowpass when height <=1
-            lo0 = batch_ifftshift(lodft)
-            lo0 = torch.ifft(lo0, signal_ndim=2, normalized=self.fft_normalize)
-            lo0_real = torch.unbind(lo0, -1)[0]
-            self.pyr_coeffs['residual_lowpass'] = lo0_real
-            self.pyr_size['residual_lowpass'] = tuple(lo0_real.shape[-2:])
-
-        if not self.downsample:
-            return self.convert_pyr_to_tensor(self.pyr_coeffs)
-        else:
-            return self.pyr_coeffs
+            lo0 = fft.ifftshift(lodft)
+            lo0 = fft.ifft2(lo0, dim=(-2,-1), norm=self.fft_norm)
+            pyr_coeffs['residual_lowpass'] = lo0.real
+            self.pyr_size['residual_lowpass'] = tuple(lo0.real.shape[-2:])
+
+        return pyr_coeffs
 
 
-    def convert_pyr_to_tensor(self, pyr_coeffs):
+    def convert_pyr_to_tensor(self, pyr_coeffs, split_complex=False):
```

Comment:
minor point: can this be a class method? You check `self.downsample` but the same thing can be checked with a try/except when you try to concatenate across bands.